yb-voyager/cmd/analyzeSchema.go

```diff
@@ -1395,3 +1388,50 @@ func createSchemaAnalysisIterationCompletedEvent(report utils.SchemaReport) cp.S
 	result.AnalysisReport = report
 	return result
 }
+
+// getSQLInfoArrayForObjectType extracts SQL info for a given object type, handling special cases like INDEX
+func getSQLInfoArrayForObjectType(schemaDir, objType string) []sqlInfo {
+	if objType != "INDEX" {
+		filePath := utils.GetObjectFilePath(schemaDir, objType)
+		return parseSqlFileForObjectType(filePath, objType)
+	}
+
+	// Special handling for INDEX - includes partition and FTS indexes
+	var sqlInfoArr []sqlInfo
+	filePath := utils.GetObjectFilePath(schemaDir, objType)
+	sqlInfoArr = append(sqlInfoArr, parseSqlFileForObjectType(filePath, objType)...)
+
+	// Add partition indexes
+	otherFPaths := utils.GetObjectFilePath(schemaDir, "PARTITION_INDEX")
+	sqlInfoArr = append(sqlInfoArr, parseSqlFileForObjectType(otherFPaths, "PARTITION_INDEX")...)
+
+	// Add FTS indexes
+	otherFPaths = utils.GetObjectFilePath(schemaDir, "FTS_INDEX")
+	sqlInfoArr = append(sqlInfoArr, parseSqlFileForObjectType(otherFPaths, "FTS_INDEX")...)
+
+	return sqlInfoArr
+}
+
+// collectAllDDLs collects all DDL statements from the schema without any issue detection
+func collectAllDDLs(sourceDBConf *srcdb.Source, skipList []string) []string {
```

Comment:
I think you can pass the dbType in the function instead of the complete conf, as that's the only param used in that

Reply:
yeah, this is also fine, its a pointer.